clients/src/main/java/org/apache/kafka/common/config/internals/BrokerSecurityConfigs.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.config.internals;
+
+import org.apache.kafka.common.config.SaslConfigs;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Common home for broker-side security configs which need to be accessible from the libraries shared
+ * between the broker and the client.
+ *
+ * Note this is an internal API and subject to change without notice.
+ */
+public class BrokerSecurityConfigs {
+
+    public static final String PRINCIPAL_BUILDER_CLASS_CONFIG = "principal.builder.class";
+    public static final String PRINCIPAL_BUILDER_CLASS_DOC = "The fully qualified name of a class that implements the " +
+            "KafkaPrincipalBuilder interface, which is used to build the KafkaPrincipal object used during " +
+            "authorization. This config also supports the deprecated PrincipalBuilder interface which was previously " +
+            "used for connections with the SSL SecurityProtocol. If not defined, a principal derived from the client " +
+            "certificate will be used in the case of client SSL authentication; for SASL, the authentication ID will " +
+            "be used as the principal name.";
```

Comment:
+1.  Currently,  now where in the Kafka documentation we mention about ANONYMOUS user. It will be good if we mention here.

Reply:
@rajinisivaram, hmm, I'd rather us specify the details or link to a config that specifies them. With security, people often struggle so the more information we can provide, the better.

Reply:
@ijuma I think we should either provide full and correct information or do that in the security documentation if it is too much as the doc for the config. I am fine with either, but at the moment, this feels incomplete to me.

Reply:
Yes, I agree that we can improve this. I was just mentioning that I prefer the option where we add/correct instead of removing. Security is the area where we get the most questions from users and the general feedback is that people find it hard to find the information they need (even if it's there at times).

Reply:
I've added more detail to the description in the latest commit.